cocotb/decorators.py

```diff
@@ -451,15 +451,23 @@ def decorator(f):
 
 @public
 class hook(coroutine, metaclass=_decorator_helper):
-    """Decorator to mark a function as a hook for cocotb.
+    """
+    Decorator to mark a function as a hook for cocotb.
 
     Used as ``@cocotb.hook()``.
 
     All hooks are run at the beginning of a cocotb test suite, prior to any
-    test code being run."""
+    test code being run.
+
+    .. deprecated:: 1.5
+        Hooks are deprecated, and the functionality can be replaced with normal module-level Python code.
```

Comment:
Is this true? I though you could run arbitrary coroutines from hooks, even if it didn't work well at all.

Reply:
Perhaps we should give an example somewhere of how to do hook-like things, such as
```python
def with_reset(test_func):
    @functools.wraps(test_func)
    def wrapped(dut):
        dut.rst <= 1
        await Timer(10, "ns")
        dut.rst <= 0
        await RisingEdge(dut.clk)
        await test_func(dut)
    return wrapped

@cocotb.test()
@with_reset
async def my_test(dut):
    # dut is already reset and ready to go!
```

Reply:
I guess I'm still not sure what the whole point of hooks were, so I had a hard time coming up with an alternative.

Reply:
Yeah, I'm not sure either. I found hooks when looking for a way to do the above, but decided that they weren't actually useful for that. I don't think we should give the above example in the docs for `hook`s, but I think it would be good tutorial content.